### PR TITLE
[libiconv] Fix arm64-ios build

### PR DESCRIPTION
**Pull request by cosmin42**
Fixes the arm64-ios build.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] SHA512s are updated for each updated download.
- [ ] The "supports" clause reflects platforms that may be fixed by this new version.
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [ ] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [ ] Only one version is added to each modified port's versions file.

Original issue:
https://github.com/microsoft/vcpkg/issues/36182

Original fix: https://git.savannah.gnu.org/gitweb/?p=gnulib.git;a=commitdiff;h=2d1edb0bd8035200d09ba4bd6d0161f37e0ee1f4

### CODE REVIEW

**Comment by cosmin42** [comment]
@microsoft-github-policy-service agree

File: ports/libiconv/portfile.cmake [hunk 1]
```diff
@@ -19,6 +19,7 @@ vcpkg_extract_source_archive(SOURCE_PATH
         0003-Add-export.patch
         0004-ModuleFileName.patch
         clang-fortify.patch # ported from https://git.savannah.gnu.org/cgit/gnulib.git/commit/?id=522aea1093a598246346b3e1c426505c344fe19a
+        0005-iOS-missing-symbol-fix.patch
```
Comment (Mengna-Li) [inline ports/libiconv/portfile.cmake:22]: Please upgrade the port-version.

**Review by Mengna-Li — CHANGES_REQUESTED** [review]

File: ports/libiconv/portfile.cmake (hunk 1 above)
Comment (cosmin42) [reply ports/libiconv/portfile.cmake:22]: Done, thank you!

**Comment by cosmin42** [comment]
@Mengna-Li  Kind reminder to review this, thank you!

**Comment by BillyONeal** [comment]
> @Mengna-Li Kind reminder to review this, thank you!

I'm pretty sure nobody was looking to review it because the build was red.

**Review by BillyONeal — APPROVED** [review]

**Review by Mengna-Li — APPROVED** [review]

**Comment by davidbien** [comment]
I've been waiting on the fix because boost-locale depends on it (and tesseract that I'd also like to use in iOS). Unless I screwed up then the build is still broken - please let me know if I somehow screwed up - but I see your patch being applied so I think I am ok. I did a git pull, I tried vcpkg update (but that just said to git pull), and I ran the bootstrap sh. 

This looks like the exact same problem I was having before. 

However, when I build with cosmin42's change via this command, then libiconv builds, but boost-locale still doesn't.
vcpkg install --triplet arm64-ios --overlay-ports=/Users/davidbien/dv/vcpkg-cosmin42/ports/libiconv

This is the output of normal vcpkg install after Billy's change - libiconv build still failing.
[cmake] Installing 1/2 libiconv:arm64-ios@1.17#3...
[cmake] Building libiconv:arm64-ios@1.17#3...
[cmake] /Users/davidbien/devenv/vcpkg/triplets/community/arm64-ios.cmake: info: loaded community triplet from here. Community triplets are not built in the curated registry and are thus less likely to succeed.
[cmake] /Users/davidbien/devenv/vcpkg/buildtrees/versioning_/versions/libiconv/0d6e7ebdbfeca59f047e66f632d1d38c2659fc14: info: installing overlay port from here
[cmake] -- Using cached libiconv-1.17.tar.gz.
[cmake] -- Cleaning sources at /Users/davidbien/devenv/vcpkg/buildtrees/libiconv/src/v1.17-84a471678e.clean. Use --editable to skip cleaning for the packages you specify.
[cmake] -- Extracting source /Users/davidbien/devenv/vcpkg/downloads/libiconv-1.17.tar.gz
[cmake] -- Applying patch 0002-Config-for-MSVC.patch
[cmake] -- Applying patch 0003-Add-export.patch
[cmake] -- Applying patch 0004-ModuleFileName.patch
[cmake] -- Applying patch clang-fortify.patch
[cmake] -- Using source at /Users/davidbien/devenv/vcpkg/buildtrees/libiconv/src/v1.17-84a471678e.clean
[cmake] -- Found external ninja('1.12.1').
[cmake] -- Getting CMake variables for arm64-ios-dbg
[cmake] -- Getting CMake variables for arm64-ios-rel
[cmake] -- Configuring arm64-ios-dbg
[cmake] -- Configuring arm64-ios-rel
[cmake] -- Building arm64-ios-dbg
[cmake] -- Installing arm64-ios-dbg
[cmake] CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
[cmake]     Command failed: /usr/bin/make -j 13 -f Makefile install DESTDIR=/Users/davidbien/devenv/vcpkg/packages/libiconv_arm64-ios
[cmake]     Working Directory: /Users/davidbien/devenv/vcpkg/buildtrees/libiconv/arm64-ios-dbg/
[cmake]     See logs for more information:
[cmake]       /Users/davidbien/devenv/vcpkg/buildtrees/libiconv/install-arm64-ios-dbg-out.log
[cmake]       /Users/davidbien/devenv/vcpkg/buildtrees/libiconv/install-arm64-ios-dbg-err.log
[cmake] 
[cmake] Call Stack (most recent call first):
[cmake]   scripts/cmake/vcpkg_build_make.cmake:153 (vcpkg_execute_build_process)
[cmake]   scripts/cmake/vcpkg_install_make.cmake:2 (vcpkg_build_make)
[cmake]   buildtrees/versioning_/versions/libiconv/0d6e7ebdbfeca59f047e66f632d1d38c2659fc14/portfile.cmake:39 (vcpkg_install_make)
[cmake]   scripts/ports.cmake:196 (include)

From the error log referenced above:
libtool: warning: remember to run 'libtool --finish /Users/davidbien/dv/sdo1/Sodorkyou/build/vcpkg_installed/arm64-ios/debug/lib'
libtool: warning: remember to run 'libtool --finish /Users/davidbien/dv/sdo1/Sodorkyou/build/vcpkg_installed/arm64-ios/debug/lib'
libtool: warning: library '/Users/davidbien/devenv/vcpkg/packages/libiconv_arm64-ios/Users/davidbien/dv/sdo1/Sodorkyou/build/vcpkg_installed/arm64-ios/debug/lib/libiconv.la' was moved.
libtool: warning: library '/Users/davidbien/devenv/vcpkg/packages/libiconv_arm64-ios/Users/davidbien/dv/sdo1/Sodorkyou/build/vcpkg_installed/arm64-ios/debug/lib/libiconv.la' was moved.
Undefined symbols for architecture arm64:
  "_stat_time_normalize", referenced from:
      _rpl_stat in stat-740bc8.o
ld: symbol(s) not found for architecture arm64
cc: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [install] Error 1
make: *** [install] Error 2

This is the output of "vcpkg install --triplet arm64-ios --overlay-ports=/Users/davidbien/dv/vcpkg-cosmin42/ports/libiconv" with cosmin42's change:
1) libiconv builds successfully.
2) boost-locale fails to build, output:

Building boost-locale:arm64-ios@1.84.0...
/Users/davidbien/devenv/vcpkg/triplets/community/arm64-ios.cmake: info: loaded community triplet from here. Community triplets are not built in the curated registry and are thus less likely to succeed.
/Users/davidbien/devenv/vcpkg/buildtrees/versioning_/versions/boost-locale/890a580576001c650e84c6b92fd995a048317e3e: info: installing overlay port from here
-- Using cached boostorg-locale-boost-1.84.0.tar.gz.
-- Cleaning sources at /Users/davidbien/devenv/vcpkg/buildtrees/boost-locale/src/ost-1.84.0-c35c322caf.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/davidbien/devenv/vcpkg/downloads/boostorg-locale-boost-1.84.0.tar.gz
-- Applying patch fix-dependencies.patch
-- Using source at /Users/davidbien/devenv/vcpkg/buildtrees/boost-locale/src/ost-1.84.0-c35c322caf.clean
-- Including /Users/davidbien/devenv/vcpkg/buildtrees/boost-locale/vcpkg-b2-options.cmake
-- Found external ninja('1.12.1').
-- Getting CMake variables for arm64-ios
-- Configuring arm64-ios
-- Building arm64-ios-dbg
-- Building arm64-ios-rel
CMake Error at /Users/davidbien/dv/sdo1/Sodorkyou/vcpkg_installed/arm64-osx/share/boost-build/boost-modular-build.cmake:196 (message):
  No libraries were produced.  This indicates a failure while building the
  boost library.
Call Stack (most recent call first):
  buildtrees/versioning_/versions/boost-locale/890a580576001c650e84c6b92fd995a048317e3e/portfile.cmake:23 (boost_modular_build)
  scripts/ports.cmake:196 (include)

**Comment by davidbien** [comment]
ok, fresh install on my new mac mini!!!

For whatever reason the new patch 0005-iOS-missing-symbol-fix.patch isn't getting applied during the install so it fails

Command: vcpkg install --triplet arm64-ios

-- Using cached libiconv-1.17.tar.gz.
-- Cleaning sources at /Users/davidbien/devenv/vcpkg/buildtrees/libiconv/src/v1.17-84a471678e.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/davidbien/devenv/vcpkg/downloads/libiconv-1.17.tar.gz
-- Applying patch 0002-Config-for-MSVC.patch
-- Applying patch 0003-Add-export.patch
-- Applying patch 0004-ModuleFileName.patch
-- Applying patch clang-fortify.patch
-- Using source at /Users/davidbien/devenv/vcpkg/buildtrees/libiconv/src/v1.17-84a471678e.clean
-- Getting CMake variables for arm64-ios-dbg
-- Getting CMake variables for arm64-ios-rel
-- Configuring arm64-ios-dbg
-- Configuring arm64-ios-rel
-- Building arm64-ios-dbg
-- Installing arm64-ios-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /usr/bin/make -j 11 -f Makefile install DESTDIR=/Users/davidbien/devenv/vcpkg/packages/libiconv_arm64-ios
    Working Directory: /Users/davidbien/devenv/vcpkg/buildtrees/libiconv/arm64-ios-dbg/
    See logs for more information:
      /Users/davidbien/devenv/vcpkg/buildtrees/libiconv/install-arm64-ios-dbg-out.log
      /Users/davidbien/devenv/vcpkg/buildtrees/libiconv/install-arm64-ios-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_make.cmake:153 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_make.cmake:2 (vcpkg_build_make)
  buildtrees/versioning_/versions/libiconv/0d6e7ebdbfeca59f047e66f632d1d38c2659fc14/portfile.cmake:39 (vcpkg_install_make)
  scripts/ports.cmake:196 (include)

**Comment by dg0yt** [comment]
Consider updating `builtin-baseline`.

**Comment by davidbien** [comment]
THANK YOU!!! lol - I had a feeling I was "doing it wrong" but baselines never caused an issue I guess cuz all my other stuff just worked all the time without mucking with them - had no idea what they were even for TBH.
Yah, it works - nice - allows me to use tesseract in my ios app as well - this is very cool.

**Pull request merged**